test/performance/scale_from_zero_test.go

```diff
@@ -0,0 +1,285 @@
+		if err != nil {
+			log.Printf("getting filename from bucket of latest results failed %#v", err)
+		}
+		latestFilename := fmt.Sprintf("%s-latest.json", scaleFromZeroTestName)
+		err = downloadLatestResults(ctx, output, KperfFlags.ServiceAccount, KperfFlags.BucketName, objPath, latestFilename)
+		if err != nil {
+			log.Printf("downloading of latest results failed %#v", err)
+			downloadFailed = true
+		}
+		if !downloadFailed {
+			latestResult := ScaleResult{}
+			err = readResultFromJsonFile(output, latestFilename, &latestResult)
+			if err != nil {
+				t.Fatalf("Failed to parse kperf service scale reference result %#v", err)
+			}
+			d, err := os.Open(output)
+			if err != nil {
+				t.Fatalf("Failed to parse kperf service scale reference result %#v", err)
+			}
+			defer d.Close()
+
+			files, err := d.Readdir(-1)
+			if err != nil {
+				t.Fatalf("Failed to parse kperf service scale reference result %#v", err)
+			}
+			var csvFilename string
+			for _, file := range files {
+				if file.Mode().IsRegular() {
+					if filepath.Ext(file.Name()) == ".csv" && strings.Contains(file.Name(), "scaling") {
+						csvFilename = file.Name()
+						break
+					}
+				}
+			}
+
+			// Open CSV file
+			f, err := os.Open(fmt.Sprintf("%s/%s", output, csvFilename))
+			if err != nil {
+				t.Fatalf("Failed to parse kperf service scale reference result %#v", err)
+			}
+			defer f.Close()
+
+			// Read File into *lines* variable
+			lines, err := csv.NewReader(f).ReadAll()
+			if err != nil {
+				t.Fatalf("Failed to parse kperf service scale reference result %#v", err)
+			}
+			for index, line := range lines {
+				if index == 0 {
+					lines[index] = append(line, "latest_service_latency", "latest_deployment_latency")
+				} else {
+					for _, s := range latestResult.Measurment {
+						if line[0] == s.ServiceName {
+							lines[index] = append(line, fmt.Sprintf("%f", s.ServiceLatency), fmt.Sprintf("%f", s.DeploymentLatency))
+							break
+						}
+					}
+				}
+			}
+
+			ff, err := os.Create(fmt.Sprintf("%s/%s-results-combined.csv", output, scaleFromZeroTestName))
+			if err != nil {
+				t.Fatalf("failed to create csv file %s\n", err)
+			}
+			defer ff.Close()
+
+			csvWriter := csv.NewWriter(ff)
+			csvWriter.WriteAll(lines)
+			csvWriter.Flush()
+
+			err = utils.GenerateHTMLFile(fmt.Sprintf("%s/%s-results-combined.csv", output, scaleFromZeroTestName), fmt.Sprintf("%s/%s-results-combined.html", output, scaleFromZeroTestName))
+			if err != nil {
+				t.Fatalf("failed to create html file %s\n", err)
+			}
+			log.Printf("CSV combined result saved in csv file %s/scale-from-zero-results-combined.csv \n", output)
+			log.Printf("HTML combined result saved in html file %s/scale-from-zero-results-combined.html \n", output)
+		}
+	}
+}
+
+func waitForScaleToZero(ctx context.Context, client *test.Clients, namespace string, objs []*v1.Service) error {
+	g := errgroup.Group{}
+	for i := 0; i < len(objs); i++ {
+		ro := objs[i]
```

Comment:
nit: personal style preference, feel free to reject

```suggestion
	for _, ro := range objs {
		ro := ro // avoid data race
```

Reply:
makes sense